FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_ARP.c

```diff
@@ -639,20 +666,26 @@ ARPPacket_t *pxARPPacket;
 
 void FreeRTOS_ClearARP( void )
 {
-	memset( xARPCache, '\0', sizeof( xARPCache ) );
+	/* Function requires the second argument to be an int.
+	 *
+	 * MISRA rule 17.7 relaxed. The return value is not 
+	 * required to be used. Also improves readability    */
+	( void ) memset( xARPCache, (int)'\0', sizeof( xARPCache ) );
 }
 /*-----------------------------------------------------------*/
 
 #if( ipconfigHAS_PRINTF != 0 ) || ( ipconfigHAS_DEBUG_PRINTF != 0 )
 
+	/* MISRA c 2012 rule 8.7 relaxed since this function can be
```

Comment:
Is the rule "relaxed" here? 

MISRA 2012 8.7 -- https://wiki.sei.cmu.edu/confluence/pages/viewpage.action?pageId=87152278. 
```FreeRTOS_PrintARPCache``` is an API, as defined in FreeRTOS_IP.h. Maybe I'm missing something?

Reply:
The sole purpose of function `FreeRTOS_PrintARPCache` is to be used for debugging. It can be called by a user for debugging the code. MISRA rule 8.7 requires that I declare this function as static (since this function is not being used in this code base).  If I declare this function as static, users will not be able to use this function.

Reply:
What does the comment on line 679 mean?  How is this rule relaxed here?  I don't see a Coverity annotation to disable it; if I run Coverity on this code will it still emit a warning here?